src/App.tsx

```diff
@@ -21,41 +23,53 @@ function login(email: string, password: string) {
         }
       }
     `
-  })
-  .then(result => {
-    localStorage.setItem('@token', result.data.login.token);
-    console.warn(result.data.login.token);
-  })
-  .catch(error => {
-    alert(error.message);
-    localStorage.setItem('@token', '');
-  })
-
+    })
+    .then((result) => {
+      localStorage.setItem('@token', result.data.login.token);
+      console.warn('logged');
+      return true;
+    })
+    .catch((error) => {
+      console.warn(error);
+      alert(error.message);
+      return false;
+    });    
 }
 
 function App() {
   const [email, setEmail] = useState('');
   const [password, setPassword] = useState('');
+  const [isLogged, setIsLogged] = useState(false);
 
-  function handleErrors() {
+  async function handleErrors() {
     const errors = Validate(email, password);
     if (errors.length > 0) {
       alert(errors);
-    }
+    } 
     else {
-      login(email, password);
+      await login(email, password) ? setIsLogged(true) : setIsLogged(false)
     }
   }
 
   return (
     <div className='App'>
       <h1>Bem-vindo(a) à Taqtile!</h1>
-      <form style= {formStyles}>
-        <EmailInput text={email} onTextChange={ (input) => {setEmail(input)} } />
-        <PasswordInput text={password} onTextChange= { (input) => {setPassword(input)} } />
-        <SubmitButton validate={ handleErrors } />
-      </form>
-      
+      <div style={formStyles}>
+        <EmailInput
+          text={email}
+          onTextChange={(input) => {
+            setEmail(input);
+          }}
+        />
+        <PasswordInput
+          text={password}
+          onTextChange={(input) => {
+            setPassword(input);
+          }}
+        />
```

Comment:
Aqui dá pra fazer igual ta feito no botão e passar só a referência da função:

```suggestion
        <EmailInput text={email} onTextChange={setEmail} />
        <PasswordInput text={password} onTextChange={setPassword} />
```